pynvml/tests/test_nvml.py

```diff
@@ -0,0 +1,210 @@
+from pynvml.nvml import *
+import pytest
+import os
+
+# Fixture to initialize and finalize nvml
+@pytest.fixture(scope='module')
+def nvml_open(request):
+    nvml_init()
+    def nvml_close():
+        nvml_finalize()
+    request.addfinalizer(nvml_close)
+
+# Get GPU count
+@pytest.fixture
+def get_gpu_count(nvml_open):
```

Comment:
It would be good to also change these fixtures as well so that they return values rather than modify the `pytest` module, as was done in the other testing file.